src/modules/launcher/PowerLauncher/App.xaml.cs

```diff
@@ -115,6 +121,24 @@ private void RegisterExitEvents()
             Current.SessionEnding += (s, e) => Dispose();
         }
 
+        [DllImport("kernel32.dll", SetLastError = true)]
+        private static extern uint WaitForSingleObject(IntPtr hHandle, uint dwMilliseconds);
+
+        private void WaitForPowerToys()
+        {
+            new Thread(() =>
+            {
+                const uint INFINITE = 0xFFFFFFFF;
+                const uint WAIT_OBJECT_0 = 0x00000000;
+
+                IntPtr powerToysProcHandle = Process.GetProcessById(_powerToysPid).Handle;
+                if (WaitForSingleObject(powerToysProcHandle, INFINITE) == WAIT_OBJECT_0)
+                {
+                    Environment.Exit(0);

```

Comment:
Would it be possible to instead send some kind of signal to the main thread to exit, instead of terminating the app from this thread?